algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -63,8 +61,26 @@ object QTree {
   def apply(k: Double): QTree[Double] = apply(k -> k)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
```

Comment:
Ah I see what you were doing here, I hadn't tweaked that the main effect of this change is to make a public field on the Semigroup that gives access to the monoid so you can pull it easily out later. Makes sense.